kdtree/include/pcl/kdtree/kdtree.h

```diff
@@ -137,7 +138,14 @@ namespace pcl
       nearestKSearch (const PointT &p_q, int k, 
                       std::vector<int> &k_indices, std::vector<float> &k_sqr_distances) const = 0;
 
-      /** \brief Search for k-nearest neighbors for the given query point.
+      template <typename Executor>
+      int
+      nearestKSearch (const Executor &exec, const PointT &p_q, int k,
+                      std::vector<int> &k_indices, std::vector<float> &k_sqr_distances) const {
+        return static_cast<Derived&>(*this).nearestKSearch(exec, p_q, k, k_indices, k_sqr_distances);
```

Comment:
This is potentially confusing IMO. You're hiding the function in the derived class without overloading. Some note somewhere detailing what's happening and more importantly why it's happening the way it is